### PR TITLE
Trunk 5331

### PR DESCRIPTION
**Pull request by saidrobley**
TRUNK-5331 CohortMembership should not require a startDate
The intersect method of the Cohort class which takes two (cohort a, cohort b) arguments have the behavior of such that the cohortMemberships that exist in both is kept, then we checked if these cohortMembership is already present. I have also changed the restriction of the startDate in order to allow it be null.

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5331

## Checklist: I completed these to help reviewers :)

- [ x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/test/java/org/openmrs/CohortTest.java [hunk 1]
```diff
@@ -165,4 +168,34 @@ public void setMemberIds_shouldSupportLargeCohorts() {
...
+
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-01-01 00:00:00");
+		Date startDate2 = dateFormat.parse("20017-02-15 00:00:00");
+
+		Cohort cohortA = new Cohort(3);
+		Cohort cohortB = new Cohort (4);
+
+		CohortMembership membershipAOne = new CohortMembership(7, startDate);
+		CohortMembership membershipATwo = new CohortMembership(7, startDate2);
```
Comment (djazayeri) [inline api/src/test/java/org/openmrs/CohortTest.java]: This seems like an invalid state to me. You should be allowed to be in the same cohort twice, but not with overlapping dates. E.g. you'd need to end membershipAOne before starting membershipATwo.

**Comment by saidrobley** [comment]
Thanks Darius Jazayeri for the review, I have made the changes you requested. Let me know if there are more changes needed.

**Comment by dkayiwa** [comment]
@saidrobley did you get a chance to look at this? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

File: api/src/test/java/org/openmrs/CohortTest.java [hunk 2]
```diff
@@ -165,4 +168,43 @@ public void setMemberIds_shouldSupportLargeCohorts() {
         double secondsToSet = (endTime - startTime)/1000;
         Assert.assertTrue("Setting cohort of size " + cohortSize + " took " + secondsToSet + " seconds", secondsToSet < 5);
     }
+
+	@Test
+	public void intersect_shouldNotContainDuplicatePatientIDs() throws Exception {
+
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-01-01 00:00:00");
+		Date startDate2 = dateFormat.parse("20017-02-15 00:00:00");
```
Comment (djazayeri) [inline api/src/test/java/org/openmrs/CohortTest.java]: this can't be right

**Comment by djazayeri** [comment]
I commented on the ticket, but will copy that here:
it's necessary to have a larger number of tests, covering multiple edge cases of the intersect method.

**Comment by dkayiwa** [comment]
@saidrobley did you see the above comments?
